core-web/libs/sdk/analytics/src/lib/react/hook/useContentAnalytics.ts

```diff
@@ -1,33 +1,44 @@
-import { useCallback, useRef } from 'react';
+import { useCallback, useMemo } from 'react';
 
 import { getUVEState } from '@dotcms/uve';
 
-import {
-    DotCMSAnalytics,
-    DotCMSAnalyticsConfig
-} from '../../core/shared/dot-content-analytics.model';
+import { DotCMSAnalytics, DotCMSAnalyticsConfig } from '../../core/shared/models';
 import { initializeAnalytics } from '../internal';
 
 /**
  * Custom hook that handles analytics tracking for anonymous users.
- * Provides methods to track events and page views with automatic timestamp injection.
- * Automatically disables tracking when inside the UVE editor.
+ * Provides methods to track events and page views.
+ *
+ * **UVE Editor Behavior:**
+ * - Automatically disables ALL tracking when inside the Universal Visual Editor (UVE)
+ * - This prevents editor interactions and preview activities from polluting analytics data
+ * - UVE detection is memoized for performance - checked once per component lifecycle
+ *
+ * **Performance:**
+ * - Uses a singleton pattern - all components with the same config share the same analytics instance
+ * - Only re-initializes when server or siteKey changes (not debug flag)
+ * - UVE state check is memoized to avoid repeated function calls
```

Comment:
This is a doc that our users will read; it should tell them how to use this, not explain how we built it.